src/main/scala/org/allenai/ike/IkeBatchSearch.scala

```diff
@@ -0,0 +1,338 @@
+    // `cat ~ 10` which should return the 10 closest word2vec neighbors to `cat`.
+    val word2vecPhrasesSearcher =
+      new EmbeddingBasedPhrasesSearcher(appConfig[Config]("word2vecPhrasesSearcher"))
+
+    // Create a table expander. This enables processing queries involving word2vec neighbors
+    // for a specified table column's contents. For e.g., if we have a MaterialConductivity
+    // table with an 'Energy' column that has some seed entries like "light" and "heat" and
+    // we would like to find similar words like "thermal energy" and "electricity", the
+    // `tableExpander` will compute the word2vec centroid of the specified column's current
+    // entries and pull up n closest neighbors. The query that would trigger this will look
+    // like: `$MaterialConductivity.Energy ~ 10`.
+    val tableExpander = new SimilarPhrasesBasedTableExpander(word2vecPhrasesSearcher)
+
+    /** Search for a pattern with a pre-created @Searcher and save the results to the given file.
+      * Results are saved as JSON-serialized @BlackLabResult objects.
+      *
+      * @param searcher the searcher to query with
+      * @param pattern the pattern to search
+      * @param outputFile the file results should be saved in
+      */
+    def searchForPattern(searcher: Searcher, pattern: NamedPattern, outputFile: File): Unit = {
+      SearchApp.parse(pattern.pattern) match {
+        case Success(query) =>
+          // Get a simplified (ready-to-execute) version of the query.
+          val interpolatedQuery = QueryLanguage.interpolateQuery(
+            query,
+            tables,
+            patterns,
+            word2vecPhrasesSearcher,
+            tableExpander
+          ).get
+
+          val results = SearchApp.search(
+            interpolatedQuery,
+            searcher,
+            batchSearchOptions.textSource,
+            Some(WINDOW_SIZE))
+          Resource.using(new PrintWriter(outputFile)) { writer =>
+            results.foreach { result =>
+              writer.println(mapper.writeValueAsString(result))
+            }
+          }
+
+        case Failure(f) =>
+          logger.error(s"Unable to parse pattern  ${pattern.name}", f)
+      }
+    }
+
+    /** Split a pattern line into a pattern name and table name, as a tuple
+      *
+      * @param patternLine the pattern line to split
+      * @return a tuple with (pattern name, table name)
+      */
+    def splitPattern(patternLine: String): Option[(String, String)] = {
+      patternLine.split("\t").map(_.trim) match {
+        case Array(patternName, tableName, _*) => Some((patternName, tableName))
+        case _ => None
+      }
+    }
+
+    val sparkConf = new SparkConf()
+      .setAppName(this.getClass.getSimpleName)
+      .setMaster("local[*]")
+      .set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+      .set("spark.kryo.registrator", "org.allenai.ike.IkeKryoRegistrator")
```

Comment:
Not on a local machine, since shuffles aren't bounded by network traffic.  On a remote cluster, this matters more.  However, the Spark documentation says Kryo should be the default; they just didn't make that change for back-compat reasons.